CMakeLists.txt

```diff
@@ -128,6 +128,17 @@ list(APPEND CMAKE_MODULE_PATH
   ${CMAKE_CURRENT_LIST_DIR}/third_party/abseil-cpp/absl/copts/
 )
 
+#-------------------------------------------------------------------------------
+# Backend functionality configuration
+#-------------------------------------------------------------------------------
+
+option(IREE_HAL_VULKAN_EMULATE_TIMELINE_SEMAPHORE
+       "Emulates timeline semaphore with binary semaphores and fences" OFF)
```

Comment:
That's actually a good point. Agreed that choices inside the build system is not preferrable. It would require quite some large scope changes though; let me get to this later.

Reply:
Added a TODO